glusterd2/commands/peers/editmetadata.go

```diff
@@ -0,0 +1,116 @@
+package peercommands
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	"github.com/gluster/glusterd2/glusterd2/peer"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/pkg/api"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+func editMetaData(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	var req api.MetaDataEditReq
+	if err := restutils.UnmarshalRequest(r, &req); err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, err.Error(), api.ErrCodeDefault)
+		return
+	}
+	peerID := mux.Vars(r)["peerid"]
+	if peerID == "" {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Peer ID not present in request", api.ErrCodeDefault)
+		return
+	}
+
```

Comment:
Return error if metadata key starts with `_`

    for k, _ := range req.MetaData{
        if strings.HasPrefix(k, "_") {
               // Error: Restricted metadata
        }
    }